### PR TITLE
Refactor push-images script to reduce duplication

### PR DESCRIPTION
**Pull request by marcofranssen**
Can be used for both scratch and non-scratch
images.

**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

N.A. (no new functionality)

**Description of change**

Reduce duplication in push images scripts.

**Which issue this PR fixes**

Reduces scope of #3607 by merging this change first.

### CODE REVIEW

File: .github/workflows/release_build.yaml
Comment (marcofranssen) [inline .github/workflows/release_build.yaml:531]: got moved into the script for reuse.

File: .github/workflows/scripts/push-images.sh
Comment (marcofranssen) [inline .github/workflows/scripts/push-images.sh:51]: Changes the registry to ghcr.io for scratch images to remove the need for duplicate script.

File: .github/workflows/scripts/push-images.sh
Comment (marcofranssen) [inline .github/workflows/scripts/push-images.sh:61]: Applies the change in image name when pushing oidc-discovery-provider to ghcr.io. To equal the naming from the push-scratch-images script.

Comment (MarcosDY) [reply .github/workflows/scripts/push-images.sh:61]: maybe I'm missing something but in 51l you are validating that the only supported variant is `scratch`, but here expecting a `static`

**Review by MarcosDY — APPROVED** [review]

**Comment by azdagron** [comment]
I'm all about reducing duplication :) One thing to consider is that we will likely no longer publish non-scratch images for 1.6.0, which will likely happen in February. At that point we'll be dropping the non-scratch version of the script. I'm not certain that making changes to this scripts now is worth the risk. If other maintainers are good with it, I'm happy to concede and accept this improvement. The script certainly looks better!

**Comment by amartinezfayo** [comment]
> I'm all about reducing duplication :) One thing to consider is that we will likely no longer publish non-scratch images for 1.6.0, which will likely happen in February. At that point we'll be dropping the non-scratch version of the script. I'm not certain that making changes to this scripts now is worth the risk. If other maintainers are good with it, I'm happy to concede and accept this improvement. The script certainly looks better!

Thank you for your feedback, @azdagron. That's a fair point. Since this improvement would be only for a couple of months, I'm now more inclined to not merge this.
Thanks @marcofranssen for this contribution! I'm closing this.

**Comment by marcofranssen** [comment]
The whole point of this PR is to also make it easier to merge #3607 

As there it will become easier with this new version of the script.

**Comment by azdagron** [comment]
Ah, ok. That makes sense. I can't re-open this PR for whatever reason. Maybe the branch was deleted on your fork or was changed to a different ref?

**Pull request closed without merging**
